Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzFramework/Entity/EntityContext.h>
+#include <AzToolsFramework/Prefab/PrefabDomTypes.h>
+
+namespace AzToolsFramework
+{
+    namespace Prefab
+    {
+        class Instance;
+        class PrefabFocusInterface;
+        class PrefabSystemComponentInterface;
+
+        class InstanceDomGenerator
+        {
+        public:
+            AZ_RTTI(InstanceUpdateExecutor, "{07E23525-1D91-41AA-B85F-136360BD1938}");
+            AZ_CLASS_ALLOCATOR(InstanceDomGenerator, AZ::SystemAllocator, 0);
+
+            void Initialize();
+
+            bool GenerateInstanceDomAccordingToCurrentFocus(const Instance* instance, PrefabDom& instanceDom) const;
```

Comment:
This should just be called GenerateInstanceDom() and should be overriding from an 'InstanceDomGeneratorInterface' method. The idea is that classes can define their own logic on how to generate a dom for the given instance. This particular classes relies upon focus mode. But maybe we need another class that doesn't.